llvm/labs/lab2/mortina_nastya/Instrument_func_wrapper.cpp

```diff
@@ -0,0 +1,72 @@
+#include "llvm/IR/Function.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/Passes/PassBuilder.h"
+#include "llvm/Passes/PassPlugin.h"
+#include "llvm/Support/raw_ostream.h"
+
+namespace {
+struct FunctionInstrumentation : llvm::PassInfoMixin<FunctionInstrumentation> {
+  llvm::PreservedAnalyses run(llvm::Function &func,
+                              llvm::FunctionAnalysisManager &) {
+    llvm::LLVMContext &ctx = func.getContext();
+    llvm::IRBuilder<> builder(ctx);
+    llvm::Module *mod = func.getParent();
+    bool hasStart = false;
+    bool hasEnd = false;
+
+    llvm::FunctionType *funcType =
+        llvm::FunctionType::get(llvm::Type::getVoidTy(ctx), false);
+    llvm::FunctionCallee startInstr =
+        (*mod).getOrInsertFunction("start_instrument", funcType);
```

Comment:
```suggestion
        mod->getOrInsertFunction("start_instrument", funcType);
```
It looks like you can apply it here and in other places